samples/detect.v2beta1.js

```diff
@@ -428,76 +276,6 @@ async function detectIntentKnowledge(
   // [END dialogflow_detect_intent_knowledge]
 }
 
-async function detectIntentwithModelSelection(
-  projectId,
-  sessionId,
-  audioFilePath,
-  languageCode,
-  model
-) {
-  // [START dialogflow_detect_intent_with_model_selection]
-  const fs = require('fs');
-
-  // Imports the Dialogflow client library
-  const dialogflow = require('dialogflow').v2beta1;
-
-  // Instantiate a DialogFlow client.
-  const sessionClient = new dialogflow.SessionsClient();
-
-  /**
-   * TODO(developer): Uncomment the following lines before running the sample.
-   */
-  // const projectId = 'ID of GCP project associated with your Dialogflow agent';
-  // const sessionId = `user specific ID of session, e.g. 12345`;
-  // const audioFilePath = `path to local audio file, e.g. ./resources/book_a_room.wav`;
-  // const languageCode = 'BCP-47 language code, e.g. en-US';
-  // const model = `speech mode selected for given request, e.g. video, phone_call, command_and_search, default`;
-
-  // Define session path
-  const sessionPath = sessionClient.sessionPath(projectId, sessionId);
-  // Read the content of the audio file and send it as part of the request.
-  const readFile = util.promisify(fs.readFile);
-  const inputAudio = await readFile(audioFilePath);
-  const request = {
-    session: sessionPath,
-    queryInput: {
-      audioConfig: {
-        audioEncoding: `AUDIO_ENCODING_LINEAR_16`,
-        sampleRateHertz: 16000,
-        languageCode: languageCode,
-        model: model,
-      },
-    },
-    inputAudio: inputAudio,
-  };
-  // Recognizes the speech in the audio and detects its intent.
-  const responses = await sessionClient.detectIntent(request);
-  const contextClient = new dialogflow.ContextsClient();
-  const result = responses[0].queryResult;
-  console.log(`  Query: ${result.queryText}`);
-  console.log(`  Response: ${result.fulfillmentText}`);
-  if (result.intent) {
-    console.log(`  Intent: ${result.intent.displayName}`);
-  } else {
-    console.log(`  No intent matched.`);
-  }
-  const parameters = JSON.stringify(struct.decode(result.parameters));
-  console.log(`  Parameters: ${parameters}`);
-  if (result.outputContexts && result.outputContexts.length) {
-    console.log(`  Output contexts:`);
-    result.outputContexts.forEach(context => {
-      const contextId = contextClient.matchContextFromContextName(context.name);
-      const contextParameters = JSON.stringify(
-        struct.decode(context.parameters)
-      );
-      console.log(`    ${contextId}`);
-      console.log(`      lifespan: ${context.lifespanCount}`);
-      console.log(`      parameters: ${contextParameters}`);
-    });
-  }
-  // [END dialogflow_detect_intent_with_model_selection]
-}
-
 const cli = require(`yargs`)
   .demand(1)
   .options({
```

Comment:
What is happening here is that `yargs` is parsing `process.argv` and calling `getKnowledgeBase` with the variables `projectId` and `knowledgeBaseId` (parsed again from `process.argv`) if a user provides the string `getKnowledgeBase` in the CLI.

This method has been removed immediately above so the linter is flagging it correctly as a missing method. If you're going to remove these samples, you simply need to also remove the corresponding commands (any command that matches the deleted sample) from the yargs configuration.

Here are some docs on the topic: https://github.com/yargs/yargs/blob/master/docs/api.md#commandmodule

Let me know if you have any questions.

Reply:
This makes perfect sense to me. I will make these changes.

Reply:
yargs-related fixes implemented.